framework/src/modules/http_api/controllers/signatures.js

```diff
@@ -55,20 +55,31 @@ SignaturesController.postSignature = async function(context, next) {
 			});
 		}
 
-		// TODO: Need to improve error handling so that we don't
-		// need to parse the error message to determine the error type.
-		const processingError = /^Error processing signature/;
-		const badRequestBodyError = /^Invalid signature body/;
-
-		if (processingError.test(data.message)) {
-			error = new ApiError(data.message, apiCodes.PROCESSING_ERROR);
-		} else if (badRequestBodyError.test(data.message)) {
-			error = new ApiError(data.message, apiCodes.BAD_REQUEST);
+		if (data.code === apiCodes.PROCESSING_ERROR) {
+			error = new ApiError(
+				'Error processing signature',
+				apiCodes.PROCESSING_ERROR,
+				data.error
```

Comment:
isn't this `data.error` a signle Error object?
ApiError should accept array of error as third parameter?

same for below

Reply:
It has been changed to an `array` from a single object, so user can see more detailed errors from `elements`.

Reply:
would it be possible to change the property name to `data.errors`?